app/src/main/java/org/mozilla/fenix/settings/studies/StudiesView.kt

```diff
@@ -47,15 +47,40 @@ class StudiesView(
     @VisibleForTesting
     internal lateinit var adapter: StudiesAdapter
 
-    @Suppress("TooGenericExceptionCaught")
+    @Suppress("TooGenericExceptionCaught", "ApplySharedPref")
     fun bind() {
         provideStudiesTitle().text = getSwitchTitle()
         provideStudiesSwitch().isChecked = settings.isExperimentationEnabled
-        provideStudiesSwitch().setOnCheckedChangeListener { _, isChecked ->
-            settings.isExperimentationEnabled = isChecked
-            experiments.globalUserParticipation = isChecked
-            provideStudiesTitle().text = getSwitchTitle()
-            quitTheApp()
+        provideStudiesSwitch().setOnClickListener {
+            val isChecked = provideStudiesSwitch().isChecked
+            provideStudiesTitle().text = getSwitchCheckedTitle()
+            val builder = AlertDialog.Builder(context)
```

Comment:
I'm kinda surprised we don't have Dialogs in AC

Reply:
There isn't any useful functionality that we could add in AC which the Android system APIs already provide. We could have _Fenix_ dialog wrappers to ensure consistent styling or UX requirements; that could work.